pkg/server/testserver.go

```diff
@@ -500,12 +501,18 @@ func (ts *TestServer) Start(ctx context.Context) error {
 	return ts.Server.Start(ctx)
 }
 
-type dummyProtectedTSProvider struct {
+type tenantProtectedTSProvider struct {
 	protectedts.Provider
+	st *cluster.Settings
 }
 
-func (d dummyProtectedTSProvider) Protect(context.Context, *kv.Txn, *ptpb.Record) error {
-	return errors.New("fake protectedts.Provider")
+func (d tenantProtectedTSProvider) Protect(
+	ctx context.Context, txn *kv.Txn, rec *ptpb.Record,
+) error {
+	if d.st.Version.IsActive(ctx, clusterversion.EnableProtectedTimestampsForTenant) {
+		return d.Provider.Protect(ctx, txn, rec)
+	}
+	return errors.New("tenant cannot Protect timestamp until the clusterversion has been finalized")
```

Comment:
Something a bit more informative perhaps? `"%s is inactive", clusterversion.EnableProtectedTimestampsForTenant`. Also [nit] flip the conditional so the error case is handled above above and the happy path here, outdented.